internal/zero/leaser/restart.go

```diff
@@ -68,13 +66,7 @@ func restartContexts(
 			return
 		}
 
-		if time.Since(start) > bo.MaxInterval {
```

Comment:
I believe we need keep this condition; otherwise `bo.Reset` would never be called.

either the `restarterFn` should take the bo.Reset as a parameter to call on success (like in connect-mux/service.go) or we need keep a current empirical condition that restart was not requested for over `bo.MaxInterval` (indicating a normal operation) 

in practical terms, based on how this function is used up the stack, if we won't do that the following would happen: 
- you set databroker connection string that's incorrect, the `restartFn` would fail continuously and quickly get the back-off interval to max value
- now you set a correct connection string, but because we never reset this back-off, the restart won't happen until after bo.MaxInterval (~1 minute)

Reply:
This doesn't make any sense. You should never compare to the MaxInterval. I don't know why you would want to do that. You end up with a "backoff" that backs off exponentially and then goes back to 0 again in a stair step pattern. Why would you want to do this? Why would you name this "backoff" when it bares no resemblance to what an exponential backoff is supposed to be?

If all of what you say above is true then I don't think I understand the intention of any of this code and don't feel qualified to work on it. (Which is exactly what I've said in code reviews on the zero code and now it's broken and I can't fix it because its too complex)